playkit/src/main/java/com/kaltura/playkit/PKMediaEntry.java

```diff
@@ -69,43 +71,56 @@ public MediaEntryType getMediaType() {
         return mediaType;
     }
 
+    public enum MediaEntryType {
+        Vod,
+        Live,
+        Unknown
+    }
+
     @Override
     public int describeContents() {
         return 0;
     }
 
     @Override
     public void writeToParcel(Parcel dest, int flags) {
-        dest.writeString(id);
-        dest.writeLong(duration);
-        dest.writeString(mediaType.name());
-        if(metadata != null) {
-            dest.writeBundle(Utils.mapToBundle(metadata));
-        } else {
-            dest.writeBundle(new Bundle());
+        dest.writeString(this.id);
+        dest.writeString(this.name);
+        dest.writeTypedList(this.sources);
```

Comment:
writing of sources should no be changed:

if(sources != null){
      dest.writeTypedList(sources);
} else {
      dest.writeTypedList(Collections.EMPTY_LIST);
}